kahuna/public/js/preview/image.js

```diff
@@ -87,6 +90,41 @@ image.directive('uiPreviewImage', function() {
     };
 });
 
+image.directive('uiPreviewImageLarge', ['observe$', 'inject$', 'imgops',
+    function(observe$, inject$, imgops) {
+        return {
+            restrict: 'E',
+            scope: {
+                image: '=',
+                hideInfo: '=',
+                selectionMode: '='
+            },
+            // extra actions can be transcluded in
+            transclude: true,
+            template: templateLarge,
+            controller: 'uiPreviewImageCtrl',
+            controllerAs: 'ctrl',
+            bindToController: true,
+            link: function(scope, element, attrs, ctrl) {
+                ctrl.loading = false;
+                const image$ = new Rx.Subject();
+
+                const optimisedImage$ = image$.flatMap((image) => {
+                    return Rx.Observable.fromPromise(imgops.getFullScreenUri(image));
+                });
+
+                scope.$watch(() => ctrl.image.data.id, () => {
```

Comment:
It's because this means we're only checking if the actual image has changed, as opposed to the whole object, which also stores things like labels and the like. We don't want to update the visible image when these are added.